### ISSUE TITLE
Asset validation failed. A value for the key 'MinimumOSVersion' in bundle Frameworks/PingOneSignals.framework is required. Xcode 15.3 Privacy manifest issue

### ISSUE THREAD

**Issue by nboyapatimolina** [pingidentity/pingone-signals-sdk-ios#2]
Hello team,

We're experiencing issues when attempting to upload builds to TestFlight or Transporter. Could we receive assistance with these errors? I've tried using both Xcode 15.3 and the 15.4 beta version, but the same error persists. Below is the error log:

**Asset validation failed (90530)**
Invalid MinimumOSVersion. Apps that only support 64-bit devices must specify a deployment target of 8.0 or later. MinimumOSVersion in ‘AppName-AppStore.app/Frameworks/PingOneSignals.framework' is ''. 

**Asset validation failed (90056)**
This bundle Payload/AppName-AppStore.app/Frameworks/PingOneSignals.framework is invalid. The Info.plist file is missing the required key: CFBundleVersion. Please find more information about CFBundleVersion at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion 

**Asset validation failed (90360)**
Missing Info.plist value. A value for the key 'MinimumOSVersion' in bundle AppName-AppStore.app/Frameworks/PingOneSignals.framework is required. 

**Asset validation failed (90208)**
Invalid Bundle. The bundle AppName-AppStore.app/Frameworks/PingOneSignals.framework does not support the minimum OS Version specified in the Info.plist. 

**Asset validation failed (90057)**
The bundle 'Payload/AppName-AppStore.app/Frameworks/PingOneSignals.framework' is missing plist key. The Info.plist file is missing the required key: CFBundleShortVersionString. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring (ID: c50f5c59-b3d0-4f5c-9583-7382d725be41)

**Comment by yizharkatif** [comment]
Thanks for reaching out, @nboyapatimolina, we'll add those keys to the Plist in next version.

In the meantime if it's urgent you can consider doing it manually - download the framework pointed out by the Package and add the required keys (MinimumOSVersion, CFBundleVersion, CFBundleShortVersionString) to the Plist file within the framework. This requires embedding the framework in the bundle instead of using SPM, of course temporary till 5.2.4 will be available.

**Comment by yizharkatif** [comment]
@nboyapatimolina - 5.2.4 had been released, and should fix this issue.

**Comment by learnwithgabbar** [comment]
Invalid Bundle. The bundle XYZ.app/Frameworks/PingOneSignals.framework does not support the minimum OS Version specified in the Info.plist. with latest release

**Comment by jcnes** [comment]
Same issue as @learnwithgabbar with 5.2.4.

**Comment by TheNightmanCodeth** [comment]
+1

**Comment by yizharkatif** [comment]
@learnwithgabbar,  @jcnes, @TheNightmanCodeth -  what's your minimum OS version?

**Comment by TheNightmanCodeth** [comment]
iOS 16

**Comment by TheNightmanCodeth** [comment]
I pulled the xcframework from the URL in the package manifest and verified `CFBundleShortVersionString` and `CFBundleVersion` are not defined in the frameworks `Info.plist`. Will try adding them manually and lyk

**Comment by jcnes** [comment]
> I pulled the xcframework from the URL in the package manifest and verified `CFBundleShortVersionString` and `CFBundleVersion` are not defined in the frameworks `Info.plist`. Will try adding them manually and lyk

+1

**Comment by yizharkatif** [comment]
Thanks @TheNightmanCodeth - the MinimumOSVersion in the Plist of PingOneSignals is iOS 12.0. Maybe your podfile specifies lower minimum version?

**Comment by TheNightmanCodeth** [comment]
@yizharkatif Using SPM for this so not sure it matters but podfile reads:

```
DEPLOYMENT_TARGET = '16.0'

platform :ios, DEPLOYMENT_TARGET
```

**Comment by yizharkatif** [comment]
Thanks @TheNightmanCodeth - SPM fails also? Where exactly does it fails: when you build, archive, or upload to App Store / TestFlight? If you could post the error message as well, it will help.

**Comment by TheNightmanCodeth** [comment]
@yizharkatif it fails during upload to appstoreconnect. 

```
2024-05-20 10:42:05.664 ERROR: [ContentDelivery.Uploader] Asset validation failed (90208) Invalid Bundle. The bundle MyApp.app/Frameworks/PingOneSignals.framework does not support the minimum OS Version specified in the Info.plist. (ID: 49ce2cd0-30d0-41be-aadc-2b171ca3f8e7)

2024-05-20 10:42:05.664 DEBUG: [ContentDelivery.Uploader] Error Domain=ContentDelivery Code=90208 "Asset validation failed" UserInfo={NSLocalizedFailureReason=Invalid Bundle. The bundle MyApp.app/Frameworks/PingOneSignals.framework does not support the minimum OS Version specified in the Info.plist. (ID: 49ce2cd0-30d0-41be-aadc-2b171ca3f8e7), NSUnderlyingError=0x600002f52b80 {Error Domain=IrisAPI Code=-19241 "Asset validation failed" UserInfo={status=409, detail=Invalid Bundle. The bundle MyApp.app/Frameworks/PingOneSignals.framework does not support the minimum OS Version specified in the Info.plist., id=49ce2cd0-30d0-41be-aadc-2b171ca3f8e7, code=STATE_ERROR.VALIDATION_ERROR.90208, title=Asset validation failed, NSLocalizedFailureReason=Invalid Bundle. The bundle MyApp.app/Frameworks/PingOneSignals.framework does not support the minimum OS Version specified in the Info.plist., NSLocalizedDescription=Asset validation failed}}, iris-code=STATE_ERROR.VALIDATION_ERROR.90208, NSLocalizedDescription=Asset validation failed}
```

**Comment by yizharkatif** [comment]
Got it @TheNightmanCodeth - fixing it ASAP. Thanks!

**Comment by yizharkatif** [comment]
@TheNightmanCodeth please test v5.2.5 (latest) on SPM.

On CocoaPods 5.2.5 will be available later on - propagation time takes several hours usually.

**Comment by TheNightmanCodeth** [comment]
@yizharkatif latest version does not resolve the issue, unfortunately. I am still seeing the same validation error from appstoreconnect

**Comment by yizharkatif** [comment]
Thanks @TheNightmanCodeth for the prompt response! I'll check again.

**Comment by TheNightmanCodeth** [comment]
@yizharkatif if it helps at all, I'm able to get around the issue by changing the minimum OS version in the frameworks Info.plist to 16.0

**Comment by yizharkatif** [comment]
It helps a lot, @TheNightmanCodeth - we've started out by missing minimum OS, and once I've added it then it conflicted with the app minimum OS version.

I'll check what should be done... Thanks!

**Comment by yizharkatif** [comment]
@TheNightmanCodeth I've just downloaded some popular iOS SDK frameworks, and they all had the MinimumOSVersion lower than 16.0 of course. Wonder why this one generates this issue.

FirebaseCore Info.plist for instance - 
![Screenshot 2024-05-20 at 23 04 33](https://github.com/pingidentity/pingone-signals-sdk-ios/assets/138602732/d0f65cae-1e19-48a0-8a53-03b668428261)

**Comment by TheNightmanCodeth** [comment]
@yizharkatif yeah it's definitely strange. I've never seen anything like it before.

**Comment by yizharkatif** [comment]
@TheNightmanCodeth if you could try 5.2.6. If it doesn't work - then I'm preparing an infrastructure to reproduce the issue and 5.2.7 will surely fix it. Thanks!

(5.2.6 is only available on SPM, not CocoaPods yet.)

**Comment by learnwithgabbar** [comment]
@yizharkatif still not working with 5.2.6

**Comment by TheNightmanCodeth** [comment]
@yizharkatif I'm giving it a shot locally but it sounds like maybe the issue is not resolved. After a bit of digging last night I'm wondering if maybe there is some platform APIs y'all are accessing in code that are not supported on iOS 12 and therefore the minimum OS version needs to be bumped up to match

**Comment by learnwithgabbar** [comment]
@yizharkatif https://forums.developer.apple.com/forums/thread/749554


![Screenshot 2024-05-21 at 19 18 22](https://github.com/pingidentity/pingone-signals-sdk-ios/assets/18284379/cf8e5164-1a9e-43fd-bb38-8d53b3fc6649)

**Comment by TheNightmanCodeth** [comment]
Confirmed still not working @yizharkatif

**Comment by yizharkatif** [comment]
Thanks @TheNightmanCodeth & @learnwithgabbar, I've reproduced the problem finally, and indeed your post @learnwithgabbar seems to be correct - I'm producing now 5.2.7 with a large number of MinimumOSVersion.

Wonder why official FirebaseCore online is still with a low number - https://github.com/pingidentity/pingone-signals-sdk-ios/issues/2#issuecomment-2121121119

**Comment by TheNightmanCodeth** [comment]
Thank you, @yizharkatif

**Comment by yizharkatif** [comment]
5.2.7 is ready, I've validated it worked around the issue.
